tests/k8st/tests/test_simple.py

```diff
@@ -0,0 +1,134 @@
+# limitations under the License.
+
+import logging
+import subprocess
+from time import sleep
+
+from kubernetes import client
+
+from tests.k8st.test_base import TestBase
+from tests.k8st.utils.utils import retry_until_success
+
+_log = logging.getLogger(__name__)
+_log.setLevel(logging.INFO)
+
+
+class TestAllRunning(TestBase):
+    def test_kubesystem_pods_running(self):
+        self.check_pod_status('kube-system')
+
+    def test_default_pods_running(self):
+        self.check_pod_status('default')
+
+    def test_calico_monitoring_pods_running(self):
+        self.check_pod_status('calico-monitoring')
+
+
+class TestSimplePolicy(TestBase):
+    def setUp(self):
+        self.create_service("nginx:1.7.9", "nginx", "policy-demo", 80)
+
+    def tearDown(self):
+        # Delete deployment
+        self.delete_and_confirm("policy-demo", "ns")
+
+    def test_simple_policy(self):
+        # Check we can talk to service.
+        retry_until_success(self.can_connect, retries=10, wait_time=1, function_args=["access"])
+
+        # Create default-deny policy
+        policy = client.V1NetworkPolicy(
+            metadata=client.V1ObjectMeta(
+                name="default-deny",
+                namespace="policy-demo"
+            ),
+            spec={
+                "podSelector": {
+                    "matchLabels": {},
+                },
+            }
+        )
+        client.ExtensionsV1beta1Api().create_namespaced_network_policy(
+            body=policy,
+            namespace="policy-demo",
+        )
+        _log.debug("Isolation policy created")
+
+        # Check we cannot talk to service
+        retry_until_success(self.cannot_connect, retries=10, wait_time=1, function_args=["access"])
+
+        # Create allow policy
+        policy = client.V1NetworkPolicy(
+            metadata=client.V1ObjectMeta(
+                name="access-nginx",
+                namespace="policy-demo"
+            ),
+            spec={
+                'ingress': [{
+                    'from': [{
+                        'podSelector': {
+                            'matchLabels': {
+                                'run': 'access'
+                            }
+                        }
+                    }]
+                }],
+                'podSelector': {
+                    'matchLabels': {
+                        'app': 'nginx'
+                    }
+                }
+            }
+        )
+        client.ExtensionsV1beta1Api().create_namespaced_network_policy(
+            body=policy,
+            namespace="policy-demo",
+        )
+        _log.debug("Allow policy created.")
+
+        # Check we can talk to service as 'access'
+        retry_until_success(self.can_connect, retries=10, wait_time=1, function_args=["access"])
+
+        # Check we cannot talk to service as 'cant-access'
+        retry_until_success(self.cannot_connect, retries=10, wait_time=1, function_args=["access"])
```

Comment:
Should this say `function-args=["cant-access"]`

Reply:
Hmm, well it's very odd that this test can be passing as is!

Reply:
Fixed now.  (After a long time understanding how it could be passing!)